uaclient/apt_news.py

```diff
@@ -210,24 +193,38 @@ def local_apt_news(cfg: UAConfig) -> bool:
             exp_dt_str = "Unknown"
         else:
             exp_dt_str = exp_dt.strftime("%d %b %Y")
-        state_files.apt_news_contents_file.write(
-            messages.CONTRACT_EXPIRED_GRACE_PERIOD_APT_NEWS.format(
-                expired_date=exp_dt_str, remaining_days=grace_period_remaining
-            )
+        return messages.CONTRACT_EXPIRED_GRACE_PERIOD_APT_NEWS.format(
+            expired_date=exp_dt_str, remaining_days=grace_period_remaining
         )
-        return True
 
     if expiry_status == ContractExpiryStatus.EXPIRED:
-        state_files.apt_news_contents_file.write(
-            messages.CONTRACT_EXPIRED_APT_NEWS
-        )
-        return True
+        return messages.CONTRACT_EXPIRED_APT_NEWS
+
+    return None
 
-    return False
+
+def format_news_for_apt_update(news: str) -> str:
+    result = "#\n"
+    for line in news.split("\n"):
+        result += "# {}\n".format(line)
+    result += "#\n"
+    return result
 
 
 def update_apt_news(cfg: UAConfig):
-    local_news_written = local_apt_news(cfg)
-    if not local_news_written:
-        apt_pkg.init()
-        fetch_and_process_apt_news(cfg)
+    try:
+        news = local_apt_news(cfg)
+        if not news:
+            apt_pkg.init()
```

Comment:
Maybe this is an overkill, but should we wrap this with `PreserveAptCfg` ? What do you think @renanrodrigo ?

Reply:
Ohhh, now that this could be called from an existing long-running process that might also use apt (e.g. update-manager, yeah we need to be more careful about this now)

Reply:
oh wait, this section of code still only gets called by `apt-news.service` so we don't need to worry about this

Reply:
yep +1 no need to worry